lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -80,26 +85,29 @@ defmodule Archethic.Mining.PendingTransactionValidation do
 
       :ok
     else
-      {:error, :invalid_tx_type} ->
-        Logger.error("Invalid Transaction Type",
+      {:error, reason} = e ->
+        Logger.info(reason,
           transaction_address: Base.encode16(address),
           transaction_type: type
         )
 
-        {:error, "Invalid Transaction Type"}
-
-      false ->
-        Logger.error("Invalid previous signature",
-          transaction_address: Base.encode16(address),
-          transaction_type: type
-        )
+        e
+    end
+  end
 
-        {:error, "Invalid previous signature"}
+  defp validate_size(%Transaction{data: data, version: tx_version}) do
+    tx_size =
+      data
+      |> TransactionData.serialize(tx_version)
+      |> byte_size()
 
-      {:error, reason} = e ->
-        Logger.info(reason, transaction_address: Base.encode16(address), transaction_type: type)
-        e
+    if tx_size >= @tx_max_size do
+      {:error, "invalid transaction: transaction data exceeds limit"}
+    else
+      :ok
     end
+  rescue
+    _ -> {:error, "invalid transaction: serialization error"}
```

Comment:
I think you can omit this rescue statement. 
The transaction would not have been transmitted at first place, if the serialization was not good